app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt

```diff
@@ -136,9 +137,7 @@ class EarnViewModel(val taskRepository: TasksRepository, val wallet: Wallet,
     }
 
     private fun isTaskAvailable(): Boolean {
-        val taskDate: Long = taskRepository.task?.startDateInMillis()
-                ?: scheduler.currentTimeMillis()
-        return scheduler.currentTimeMillis() >= taskDate
+        return taskRepository.isTaskAvailable()
```

Comment:
Why do we need this method

Reply:
you are right - will remove the method and call taskRepository.isTaskAvailable directly